pkg/sensors/handler.go

```diff
@@ -6,33 +6,70 @@ package sensors
 import (
 	"fmt"
 
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/policyfilter"
 	"github.com/cilium/tetragon/pkg/tracingpolicy"
 )
 
 type handler struct {
 	// map of sensor collections: name -> collection
 	collections               map[string]collection
 	bpfDir, mapDir, ciliumDir string
+
+	nextPolicyID uint64
+	pfState      *policyfilter.State
 }
 
-func newHandler(bpfDir, mapDir, ciliumDir string) *handler {
+func newHandler(bpfDir, mapDir, ciliumDir string) (*handler, error) {
+	pfState, err := policyfilter.GetState()
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize policy filter state: %w", err)
+	}
+
 	return &handler{
 		collections: map[string]collection{},
 		bpfDir:      bpfDir,
 		mapDir:      mapDir,
 		ciliumDir:   ciliumDir,
-	}
+		pfState:     pfState,
+		// NB: policy ids start with 1, so that they can be used for filtering. In policy
+		// filtering code, a policy id of 0 means no filtering.
+		nextPolicyID: 1,
+	}, nil
+}
+
+func (h *handler) allocPolicyID() uint64 {
+	ret := h.nextPolicyID
+	h.nextPolicyID++
```

Comment:
What about you make it more explicit , in newHandler() you do:

```
return &handler{ nextPolicyID: ZERO, }
```

and in allocPolicyID() you increment then return the usual way... per this logic I suppose you have to call allocPolicyID() anyway.

Then you may even convert that to `atomic.AddUint64() ` to be robust

Reply:
What is the problem with the current approach?